### PR TITLE
equals and hashcode

### PR DESCRIPTION
**Pull request by Vinglish**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -16,6 +18,22 @@ public Rectangle(Integer width, Integer length, String color) {
         this.color = color;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this != o && getClass() != o.getClass()) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: what about o == null?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -16,6 +18,22 @@ public Rectangle(Integer width, Integer length, String color) {
...
+        }
+        Rectangle rectangle = (Rectangle) o;
+        return Objects.equals(width, rectangle.width)
+                && Objects.equals(length, rectangle.length)
+                && Objects.equals(color, rectangle.color);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(width, length, color);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: write your own implementation using primary number 31

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (Vinglish) [reply src/main/java/core/basesyntax/Rectangle.java]: I tried to do this but InteliJ told me "this != o" covered "o == null".  to add anyway?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -16,6 +18,22 @@ public Rectangle(Integer width, Integer length, String color) {
         this.color = color;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this != o && getClass() != o.getClass()) {
+            return false;
+        }
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java:28]: let's check if (this == o) return true;

File: src/main/java/core/basesyntax/Rectangle.java (hunk 2 above)
Comment (Vinglish) [reply src/main/java/core/basesyntax/Rectangle.java]: Done

File: src/main/java/core/basesyntax/Rectangle.java (hunk 3 above)
Comment (Vinglish) [reply src/main/java/core/basesyntax/Rectangle.java:28]: Done

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**